packages/nx-extensions/src/executors/assert-built-esm-and-cjs/executor.ts

```diff
@@ -0,0 +1,57 @@
+import { ExecutorContext } from '@nrwl/devkit';
+import { spawnSync } from 'child_process';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import * as path from 'path';
+import { AssertBuiltEsmAndCjsExecutorSchema } from './schema';
+
+/**
+ * Test whether a module can be imported as both ESM and CJS.
+ *
+ * @param options
+ * @param context
+ * @returns
+ */
+export default async function runExecutor(
+	options: AssertBuiltEsmAndCjsExecutorSchema,
+	context: ExecutorContext
+) {
+	const buildDir = options.outputPath.split('/')[0];
+	const testsPath = path.join(context.root, buildDir, 'test-esm-cjs');
+	if (!existsSync(testsPath)) {
+		mkdirSync(testsPath);
+	}
+
+	writeFileSync(
+		path.join(testsPath, 'test-esm.mjs'),
+		`import * as result from '../../${options.outputPath}/index.js';`
+	);
+	writeFileSync(
+		path.join(testsPath, 'test-cjs.cjs'),
+		`require('../../${options.outputPath}');`
+	);
+	writeFileSync(
+		path.join(testsPath, 'test.sh'),
+		`#!/bin/sh
+set -e
+node test-esm.mjs;
+node test-cjs.cjs;
+echo Success;`
+	);
```

Comment:
fun, I haven't seen `set -e` without `pipefail`, but it makes sense here. still, I wonder if the name of this file could be clearer. assuming I understand this properly, something like `ensure-both-outputs-run-in-node.sh` could communicate that without needing a comment.

Reply:
Good idea, thanks!